test_pull_requests

```diff
@@ -862,8 +1058,9 @@ popd
       repo_to_pull_request.each do |repo, pull_request|
         pull_request_comments = get_comments(pull_request['number'], repo)
         test_prefix_comment = get_comment_with_prefix(pull_request['number'], repo, settings['test_prefix'], pull_request_comments)
-        if !test_prefix_comment || !(test_prefix_comment['body'] =~ /^#{settings['test_prefix']} Waiting/)
-          being_queued_comment = "#{settings['test_prefix']} Waiting: Determining build queue position"
+        fields = extract_bot_comment_fields(test_prefix_comment['body'], settings) if test_prefix_comment
+        if !test_prefix_comment || !([:wait_in_queue, :wait_queue_pos, :wait_stable_build].include? fields[:state])
```

Comment:
There is no argument that a job that was waiting on a downstream job and is now no longer waiting on that but now is waiting on the build queue should not see its comment updated to reflect that fact.